.github/workflows/ci.yml

```diff
@@ -0,0 +1,65 @@
+name: CI
+on:
+  - push
+  - pull_request
+
+jobs:
```

Comment:
can we maybe split this into 3 jobs? e.g.:
- linters
- binaries
- unit-tests

Reply:
Thank you for the work by the way!

Reply:
Okay, I'll split them into parallel jobs. Do you actually want test binaries to be saved as artifacts or that can be dropped?

Reply:
third parties might be using the binaries, so it's good to keep them.

Reply:
Split done.

Reply:
yup, saw that. thanks!